processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/TypeSelector.java

```diff
@@ -84,10 +85,22 @@ public <T extends Method> List<SelectedMethod<T>> getMatchingMethods(Method mapp
         return result;
     }
 
-    private List<ParameterBinding> getAvailableParameterBindingsFromMethod(Method method, Type targetType,
-        SourceRHS sourceRHS) {
+    private <T extends Method> List<ParameterBinding> getAvailableParameterBindingsFromMethod(
+        Method method,
+        Type targetType,
+        SourceRHS sourceRHS,
+        List<SelectedMethod<T>> methods
+    ) {
         List<ParameterBinding> availableParams = new ArrayList<>( method.getParameters().size() + 3 );
 
+        for (SelectedMethod<T> selectedMethod : methods) {
+          Parameter propertyNameParameter = selectedMethod.getMethod().getTargetPropertyNameParameter();
+          if (propertyNameParameter != null) {
+            availableParams.add( ParameterBinding
+              .fromParameter( propertyNameParameter ) );
+          }
+        }
+
```

Comment:
is this still needed?

Can't we solve this in the same manner as is done for `@TargetType` in the `addMappingTargetAndTargetTypeBindings` method?

Reply:
I'll look into it and try to solve it as you suggested. (currently its needed ...)